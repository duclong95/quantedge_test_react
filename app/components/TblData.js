import React, { Component } from 'react';
import {
    Table,
    TableBody,
    TableRowColumn,
    TableRow,
} from 'material-ui/Table';
import data from '../data/data'
import '../css/style.css'
var NumberFormat = require('react-number-format');

class TblData extends Component {

    constructor(props) {
        super(props);
        this.state = {
            dataState: data,
            sort: 'decrease'
        }
    }
    update() {
        var items = this.state.dataState;

        items.map(item => {
            var limit = item.price * (5 / 100);
            let change = (Math.random() * (limit + limit) - limit).toFixed(2);
            var percentChange = ((change / item.price) * 100).toFixed(2);
            var updateVolume = Math.random() * (30 - 10) + 10;
            item.change = change;
            item.percentChange = percentChange;
            item.price = (parseFloat(item.price) + parseFloat(change)).toFixed(2);
            item.volume = item.volume + parseInt(updateVolume, 10);
        })
        this.setState({
            dataState: items,
        })
    }

    componentWillMount() {
       setInterval(this.update.bind(this),5000);  
    }
    
    render() {
        var list = this.state.dataState.sort((a, b) => {
            if (this.state.sort === "increase") {
                return (a.price * a.volume) - (b.price * b.volume);
            }
            if (this.state.sort === "decrease") {
                return (b.price * b.volume) - (a.price * a.volume);
            }
        }).map(item => {
            var className = 'increase'
            if (item.change < 0) {
                className = 'decrease';
            }
            var value = (item.price * item.volume).toFixed(0);
            return (
                <TableRow key={item.id}>
                    <TableRowColumn style={{ width: '10%', color: '#119fe1' }}>{item.code}</TableRowColumn>
                    <TableRowColumn style={{ width: '29%', color:'#9e9e9e' }}>{item.company}</TableRowColumn>
                    <TableRowColumn style={{ width: '19%' }}>{item.price}</TableRowColumn>
                    <TableRowColumn style={{ width: '14%' }}><NumberFormat value={value} displayType={'text'} thousandSeparator={true} /></TableRowColumn>
                    <TableRowColumn className={className} style={{ width: '14%' }}>{item.change}</TableRowColumn>
                    <TableRowColumn className={className} style={{ width: '14%' }}>{item.percentChange}%</TableRowColumn>
                </TableRow>
            )
        })

        return (
            <Table>
                <TableBody displayRowCheckbox={false}>
                    {list}
                </TableBody>
            </Table>
        )
    }
}
export default TblData;    
