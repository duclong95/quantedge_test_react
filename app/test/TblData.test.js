import React from 'react'
import {shallow} from 'enzyme'
import TblData from '../components/TblData'

test('Header',()=>{
    const component = shallow(<TblData/>);
})