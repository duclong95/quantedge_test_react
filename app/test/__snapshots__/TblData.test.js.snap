// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render TblData 1`] = `
<Table
  allRowsSelected={false}
  fixedFooter={true}
  fixedHeader={true}
  height="inherit"
  multiSelectable={false}
  selectable={true}
>
  <TableBody
    allRowsSelected={false}
    deselectOnClickaway={true}
    displayRowCheckbox={false}
    multiSelectable={false}
    preScanRows={true}
    selectable={true}
    style={Object {}}
  >
    <TableRow
      displayBorder={true}
      hoverable={false}
      hovered={false}
      key="0"
      selectable={true}
      striped={false}
    >
      <TableRowColumn
        className="code"
        hoverable={false}
      >
        AAA.AX
      </TableRowColumn>
      <TableRowColumn
        className="company"
        hoverable={false}
      >
        AUSTRALIA
      </TableRowColumn>
      <TableRowColumn
        className="price"
        hoverable={false}
      >
        51.55
      </TableRowColumn>
      <TableRowColumn
        className="value"
        hoverable={false}
      >
        <NumberFormat
          allowNegative={true}
          decimalSeparator="."
          displayType="text"
          isAllowed={[Function]}
          onChange={[Function]}
          onKeyDown={[Function]}
          onMouseUp={[Function]}
          prefix=""
          suffix=""
          thousandSeparator={true}
          type="text"
          value="18076575"
        />
      </TableRowColumn>
      <TableRowColumn
        className="increase"
        hoverable={false}
        style={
          Object {
            "width": "14%",
          }
        }
      >
        0
      </TableRowColumn>
      <TableRowColumn
        className="increase"
        hoverable={false}
        style={
          Object {
            "width": "14%",
          }
        }
      >
        0
        %
      </TableRowColumn>
    </TableRow>
    <TableRow
      displayBorder={true}
      hoverable={false}
      hovered={false}
      key="1"
      selectable={true}
      striped={false}
    >
      <TableRowColumn
        className="code"
        hoverable={false}
      >
        ABE.AX
      </TableRowColumn>
      <TableRowColumn
        className="company"
        hoverable={false}
      >
        ICELANDICELAND
      </TableRowColumn>
      <TableRowColumn
        className="price"
        hoverable={false}
      >
        20.92
      </TableRowColumn>
      <TableRowColumn
        className="value"
        hoverable={false}
      >
        <NumberFormat
          allowNegative={true}
          decimalSeparator="."
          displayType="text"
          isAllowed={[Function]}
          onChange={[Function]}
          onKeyDown={[Function]}
          onMouseUp={[Function]}
          prefix=""
          suffix=""
          thousandSeparator={true}
          type="text"
          value="5303597"
        />
      </TableRowColumn>
      <TableRowColumn
        className="increase"
        hoverable={false}
        style={
          Object {
            "width": "14%",
          }
        }
      >
        0
      </TableRowColumn>
      <TableRowColumn
        className="increase"
        hoverable={false}
        style={
          Object {
            "width": "14%",
          }
        }
      >
        0
        %
      </TableRowColumn>
    </TableRow>
    <TableRow
      displayBorder={true}
      hoverable={false}
      hovered={false}
      key="2"
      selectable={true}
      striped={false}
    >
      <TableRowColumn
        className="code"
        hoverable={false}
      >
        AAC.CX
      </TableRowColumn>
      <TableRowColumn
        className="company"
        hoverable={false}
      >
        USA
      </TableRowColumn>
      <TableRowColumn
        className="price"
        hoverable={false}
      >
        38.56
      </TableRowColumn>
      <TableRowColumn
        className="value"
        hoverable={false}
      >
        <NumberFormat
          allowNegative={true}
          decimalSeparator="."
          displayType="text"
          isAllowed={[Function]}
          onChange={[Function]}
          onKeyDown={[Function]}
          onMouseUp={[Function]}
          prefix=""
          suffix=""
          thousandSeparator={true}
          type="text"
          value="29829360"
        />
      </TableRowColumn>
      <TableRowColumn
        className="increase"
        hoverable={false}
        style={
          Object {
            "width": "14%",
          }
        }
      >
        0
      </TableRowColumn>
      <TableRowColumn
        className="increase"
        hoverable={false}
        style={
          Object {
            "width": "14%",
          }
        }
      >
        0
        %
      </TableRowColumn>
    </TableRow>
    <TableRow
      displayBorder={true}
      hoverable={false}
      hovered={false}
      key="3"
      selectable={true}
      striped={false}
    >
      <TableRowColumn
        className="code"
        hoverable={false}
      >
        AAD.AX
      </TableRowColumn>
      <TableRowColumn
        className="company"
        hoverable={false}
      >
        ITALY
      </TableRowColumn>
      <TableRowColumn
        className="price"
        hoverable={false}
      >
        79.65
      </TableRowColumn>
      <TableRowColumn
        className="value"
        hoverable={false}
      >
        <NumberFormat
          allowNegative={true}
          decimalSeparator="."
          displayType="text"
          isAllowed={[Function]}
          onChange={[Function]}
          onKeyDown={[Function]}
          onMouseUp={[Function]}
          prefix=""
          suffix=""
          thousandSeparator={true}
          type="text"
          value="49881768"
        />
      </TableRowColumn>
      <TableRowColumn
        className="increase"
        hoverable={false}
        style={
          Object {
            "width": "14%",
          }
        }
      >
        0
      </TableRowColumn>
      <TableRowColumn
        className="increase"
        hoverable={false}
        style={
          Object {
            "width": "14%",
          }
        }
      >
        0
        %
      </TableRowColumn>
    </TableRow>
    <TableRow
      displayBorder={true}
      hoverable={false}
      hovered={false}
      key="4"
      selectable={true}
      striped={false}
    >
      <TableRowColumn
        className="code"
        hoverable={false}
      >
        AAE.AX
      </TableRowColumn>
      <TableRowColumn
        className="company"
        hoverable={false}
      >
        ENGLAND
      </TableRowColumn>
      <TableRowColumn
        className="price"
        hoverable={false}
      >
        88.62
      </TableRowColumn>
      <TableRowColumn
        className="value"
        hoverable={false}
      >
        <NumberFormat
          allowNegative={true}
          decimalSeparator="."
          displayType="text"
          isAllowed={[Function]}
          onChange={[Function]}
          onKeyDown={[Function]}
          onMouseUp={[Function]}
          prefix=""
          suffix=""
          thousandSeparator={true}
          type="text"
          value="77213099"
        />
      </TableRowColumn>
      <TableRowColumn
        className="increase"
        hoverable={false}
        style={
          Object {
            "width": "14%",
          }
        }
      >
        0
      </TableRowColumn>
      <TableRowColumn
        className="increase"
        hoverable={false}
        style={
          Object {
            "width": "14%",
          }
        }
      >
        0
        %
      </TableRowColumn>
    </TableRow>
    <TableRow
      displayBorder={true}
      hoverable={false}
      hovered={false}
      key="5"
      selectable={true}
      striped={false}
    >
      <TableRowColumn
        className="code"
        hoverable={false}
      >
        AAF.AX
      </TableRowColumn>
      <TableRowColumn
        className="company"
        hoverable={false}
      >
        EGYPT
      </TableRowColumn>
      <TableRowColumn
        className="price"
        hoverable={false}
      >
        98.53
      </TableRowColumn>
      <TableRowColumn
        className="value"
        hoverable={false}
      >
        <NumberFormat
          allowNegative={true}
          decimalSeparator="."
          displayType="text"
          isAllowed={[Function]}
          onChange={[Function]}
          onKeyDown={[Function]}
          onMouseUp={[Function]}
          prefix=""
          suffix=""
          thousandSeparator={true}
          type="text"
          value="56866984"
        />
      </TableRowColumn>
      <TableRowColumn
        className="increase"
        hoverable={false}
        style={
          Object {
            "width": "14%",
          }
        }
      >
        0
      </TableRowColumn>
      <TableRowColumn
        className="increase"
        hoverable={false}
        style={
          Object {
            "width": "14%",
          }
        }
      >
        0
        %
      </TableRowColumn>
    </TableRow>
    <TableRow
      displayBorder={true}
      hoverable={false}
      hovered={false}
      key="6"
      selectable={true}
      striped={false}
    >
      <TableRowColumn
        className="code"
        hoverable={false}
      >
        AAG.AX
      </TableRowColumn>
      <TableRowColumn
        className="company"
        hoverable={false}
      >
        TURKEY
      </TableRowColumn>
      <TableRowColumn
        className="price"
        hoverable={false}
      >
        51.17
      </TableRowColumn>
      <TableRowColumn
        className="value"
        hoverable={false}
      >
        <NumberFormat
          allowNegative={true}
          decimalSeparator="."
          displayType="text"
          isAllowed={[Function]}
          onChange={[Function]}
          onKeyDown={[Function]}
          onMouseUp={[Function]}
          prefix=""
          suffix=""
          thousandSeparator={true}
          type="text"
          value="26848234"
        />
      </TableRowColumn>
      <TableRowColumn
        className="increase"
        hoverable={false}
        style={
          Object {
            "width": "14%",
          }
        }
      >
        0
      </TableRowColumn>
      <TableRowColumn
        className="increase"
        hoverable={false}
        style={
          Object {
            "width": "14%",
          }
        }
      >
        0
        %
      </TableRowColumn>
    </TableRow>
    <TableRow
      displayBorder={true}
      hoverable={false}
      hovered={false}
      key="7"
      selectable={true}
      striped={false}
    >
      <TableRowColumn
        className="code"
        hoverable={false}
      >
        AAH.AX
      </TableRowColumn>
      <TableRowColumn
        className="company"
        hoverable={false}
      >
        BELGIUM
      </TableRowColumn>
      <TableRowColumn
        className="price"
        hoverable={false}
      >
        4.09
      </TableRowColumn>
      <TableRowColumn
        className="value"
        hoverable={false}
      >
        <NumberFormat
          allowNegative={true}
          decimalSeparator="."
          displayType="text"
          isAllowed={[Function]}
          onChange={[Function]}
          onKeyDown={[Function]}
          onMouseUp={[Function]}
          prefix=""
          suffix=""
          thousandSeparator={true}
          type="text"
          value="3964539"
        />
      </TableRowColumn>
      <TableRowColumn
        className="increase"
        hoverable={false}
        style={
          Object {
            "width": "14%",
          }
        }
      >
        0
      </TableRowColumn>
      <TableRowColumn
        className="increase"
        hoverable={false}
        style={
          Object {
            "width": "14%",
          }
        }
      >
        0
        %
      </TableRowColumn>
    </TableRow>
    <TableRow
      displayBorder={true}
      hoverable={false}
      hovered={false}
      key="8"
      selectable={true}
      striped={false}
    >
      <TableRowColumn
        className="code"
        hoverable={false}
      >
        AAI.AX
      </TableRowColumn>
      <TableRowColumn
        className="company"
        hoverable={false}
      >
        USA
      </TableRowColumn>
      <TableRowColumn
        className="price"
        hoverable={false}
      >
        89.96
      </TableRowColumn>
      <TableRowColumn
        className="value"
        hoverable={false}
      >
        <NumberFormat
          allowNegative={true}
          decimalSeparator="."
          displayType="text"
          isAllowed={[Function]}
          onChange={[Function]}
          onKeyDown={[Function]}
          onMouseUp={[Function]}
          prefix=""
          suffix=""
          thousandSeparator={true}
          type="text"
          value="69104393"
        />
      </TableRowColumn>
      <TableRowColumn
        className="increase"
        hoverable={false}
        style={
          Object {
            "width": "14%",
          }
        }
      >
        0
      </TableRowColumn>
      <TableRowColumn
        className="increase"
        hoverable={false}
        style={
          Object {
            "width": "14%",
          }
        }
      >
        0
        %
      </TableRowColumn>
    </TableRow>
    <TableRow
      displayBorder={true}
      hoverable={false}
      hovered={false}
      key="9"
      selectable={true}
      striped={false}
    >
      <TableRowColumn
        className="code"
        hoverable={false}
      >
        AAK.AX
      </TableRowColumn>
      <TableRowColumn
        className="company"
        hoverable={false}
      >
        GERMANY
      </TableRowColumn>
      <TableRowColumn
        className="price"
        hoverable={false}
      >
        72.35
      </TableRowColumn>
      <TableRowColumn
        className="value"
        hoverable={false}
      >
        <NumberFormat
          allowNegative={true}
          decimalSeparator="."
          displayType="text"
          isAllowed={[Function]}
          onChange={[Function]}
          onKeyDown={[Function]}
          onMouseUp={[Function]}
          prefix=""
          suffix=""
          thousandSeparator={true}
          type="text"
          value="9214930"
        />
      </TableRowColumn>
      <TableRowColumn
        className="increase"
        hoverable={false}
        style={
          Object {
            "width": "14%",
          }
        }
      >
        0
      </TableRowColumn>
      <TableRowColumn
        className="increase"
        hoverable={false}
        style={
          Object {
            "width": "14%",
          }
        }
      >
        0
        %
      </TableRowColumn>
    </TableRow>
    <TableRow
      displayBorder={true}
      hoverable={false}
      hovered={false}
      key="10"
      selectable={true}
      striped={false}
    >
      <TableRowColumn
        className="code"
        hoverable={false}
      >
        AAL.AX
      </TableRowColumn>
      <TableRowColumn
        className="company"
        hoverable={false}
      >
        SPAIN
      </TableRowColumn>
      <TableRowColumn
        className="price"
        hoverable={false}
      >
        83.53
      </TableRowColumn>
      <TableRowColumn
        className="value"
        hoverable={false}
      >
        <NumberFormat
          allowNegative={true}
          decimalSeparator="."
          displayType="text"
          isAllowed={[Function]}
          onChange={[Function]}
          onKeyDown={[Function]}
          onMouseUp={[Function]}
          prefix=""
          suffix=""
          thousandSeparator={true}
          type="text"
          value="9534448"
        />
      </TableRowColumn>
      <TableRowColumn
        className="increase"
        hoverable={false}
        style={
          Object {
            "width": "14%",
          }
        }
      >
        0
      </TableRowColumn>
      <TableRowColumn
        className="increase"
        hoverable={false}
        style={
          Object {
            "width": "14%",
          }
        }
      >
        0
        %
      </TableRowColumn>
    </TableRow>
    <TableRow
      displayBorder={true}
      hoverable={false}
      hovered={false}
      key="11"
      selectable={true}
      striped={false}
    >
      <TableRowColumn
        className="code"
        hoverable={false}
      >
        AAM.AX
      </TableRowColumn>
      <TableRowColumn
        className="company"
        hoverable={false}
      >
        CHINA
      </TableRowColumn>
      <TableRowColumn
        className="price"
        hoverable={false}
      >
        96.33
      </TableRowColumn>
      <TableRowColumn
        className="value"
        hoverable={false}
      >
        <NumberFormat
          allowNegative={true}
          decimalSeparator="."
          displayType="text"
          isAllowed={[Function]}
          onChange={[Function]}
          onKeyDown={[Function]}
          onMouseUp={[Function]}
          prefix=""
          suffix=""
          thousandSeparator={true}
          type="text"
          value="60379933"
        />
      </TableRowColumn>
      <TableRowColumn
        className="increase"
        hoverable={false}
        style={
          Object {
            "width": "14%",
          }
        }
      >
        0
      </TableRowColumn>
      <TableRowColumn
        className="increase"
        hoverable={false}
        style={
          Object {
            "width": "14%",
          }
        }
      >
        0
        %
      </TableRowColumn>
    </TableRow>
    <TableRow
      displayBorder={true}
      hoverable={false}
      hovered={false}
      key="12"
      selectable={true}
      striped={false}
    >
      <TableRowColumn
        className="code"
        hoverable={false}
      >
        AAN.AX
      </TableRowColumn>
      <TableRowColumn
        className="company"
        hoverable={false}
      >
        PHILIPPINES
      </TableRowColumn>
      <TableRowColumn
        className="price"
        hoverable={false}
      >
        53.85
      </TableRowColumn>
      <TableRowColumn
        className="value"
        hoverable={false}
      >
        <NumberFormat
          allowNegative={true}
          decimalSeparator="."
          displayType="text"
          isAllowed={[Function]}
          onChange={[Function]}
          onKeyDown={[Function]}
          onMouseUp={[Function]}
          prefix=""
          suffix=""
          thousandSeparator={true}
          type="text"
          value="51566437"
        />
      </TableRowColumn>
      <TableRowColumn
        className="increase"
        hoverable={false}
        style={
          Object {
            "width": "14%",
          }
        }
      >
        0
      </TableRowColumn>
      <TableRowColumn
        className="increase"
        hoverable={false}
        style={
          Object {
            "width": "14%",
          }
        }
      >
        0
        %
      </TableRowColumn>
    </TableRow>
    <TableRow
      displayBorder={true}
      hoverable={false}
      hovered={false}
      key="13"
      selectable={true}
      striped={false}
    >
      <TableRowColumn
        className="code"
        hoverable={false}
      >
        AAO.AX
      </TableRowColumn>
      <TableRowColumn
        className="company"
        hoverable={false}
      >
        RUSSIA
      </TableRowColumn>
      <TableRowColumn
        className="price"
        hoverable={false}
      >
        7.53
      </TableRowColumn>
      <TableRowColumn
        className="value"
        hoverable={false}
      >
        <NumberFormat
          allowNegative={true}
          decimalSeparator="."
          displayType="text"
          isAllowed={[Function]}
          onChange={[Function]}
          onKeyDown={[Function]}
          onMouseUp={[Function]}
          prefix=""
          suffix=""
          thousandSeparator={true}
          type="text"
          value="3912407"
        />
      </TableRowColumn>
      <TableRowColumn
        className="increase"
        hoverable={false}
        style={
          Object {
            "width": "14%",
          }
        }
      >
        0
      </TableRowColumn>
      <TableRowColumn
        className="increase"
        hoverable={false}
        style={
          Object {
            "width": "14%",
          }
        }
      >
        0
        %
      </TableRowColumn>
    </TableRow>
    <TableRow
      displayBorder={true}
      hoverable={false}
      hovered={false}
      key="14"
      selectable={true}
      striped={false}
    >
      <TableRowColumn
        className="code"
        hoverable={false}
      >
        AAP.AX
      </TableRowColumn>
      <TableRowColumn
        className="company"
        hoverable={false}
      >
        LAOS
      </TableRowColumn>
      <TableRowColumn
        className="price"
        hoverable={false}
      >
        26.70
      </TableRowColumn>
      <TableRowColumn
        className="value"
        hoverable={false}
      >
        <NumberFormat
          allowNegative={true}
          decimalSeparator="."
          displayType="text"
          isAllowed={[Function]}
          onChange={[Function]}
          onKeyDown={[Function]}
          onMouseUp={[Function]}
          prefix=""
          suffix=""
          thousandSeparator={true}
          type="text"
          value="18652326"
        />
      </TableRowColumn>
      <TableRowColumn
        className="increase"
        hoverable={false}
        style={
          Object {
            "width": "14%",
          }
        }
      >
        0
      </TableRowColumn>
      <TableRowColumn
        className="increase"
        hoverable={false}
        style={
          Object {
            "width": "14%",
          }
        }
      >
        0
        %
      </TableRowColumn>
    </TableRow>
    <TableRow
      displayBorder={true}
      hoverable={false}
      hovered={false}
      key="15"
      selectable={true}
      striped={false}
    >
      <TableRowColumn
        className="code"
        hoverable={false}
      >
        AAB.AX
      </TableRowColumn>
      <TableRowColumn
        className="company"
        hoverable={false}
      >
        IRAQ
      </TableRowColumn>
      <TableRowColumn
        className="price"
        hoverable={false}
      >
        21.26
      </TableRowColumn>
      <TableRowColumn
        className="value"
        hoverable={false}
      >
        <NumberFormat
          allowNegative={true}
          decimalSeparator="."
          displayType="text"
          isAllowed={[Function]}
          onChange={[Function]}
          onKeyDown={[Function]}
          onMouseUp={[Function]}
          prefix=""
          suffix=""
          thousandSeparator={true}
          type="text"
          value="4651220"
        />
      </TableRowColumn>
      <TableRowColumn
        className="increase"
        hoverable={false}
        style={
          Object {
            "width": "14%",
          }
        }
      >
        0
      </TableRowColumn>
      <TableRowColumn
        className="increase"
        hoverable={false}
        style={
          Object {
            "width": "14%",
          }
        }
      >
        0
        %
      </TableRowColumn>
    </TableRow>
    <TableRow
      displayBorder={true}
      hoverable={false}
      hovered={false}
      key="16"
      selectable={true}
      striped={false}
    >
      <TableRowColumn
        className="code"
        hoverable={false}
      >
        AAR.AX
      </TableRowColumn>
      <TableRowColumn
        className="company"
        hoverable={false}
      >
        SINGAPORE
      </TableRowColumn>
      <TableRowColumn
        className="price"
        hoverable={false}
      >
        75.79
      </TableRowColumn>
      <TableRowColumn
        className="value"
        hoverable={false}
      >
        <NumberFormat
          allowNegative={true}
          decimalSeparator="."
          displayType="text"
          isAllowed={[Function]}
          onChange={[Function]}
          onKeyDown={[Function]}
          onMouseUp={[Function]}
          prefix=""
          suffix=""
          thousandSeparator={true}
          type="text"
          value="36466662"
        />
      </TableRowColumn>
      <TableRowColumn
        className="increase"
        hoverable={false}
        style={
          Object {
            "width": "14%",
          }
        }
      >
        0
      </TableRowColumn>
      <TableRowColumn
        className="increase"
        hoverable={false}
        style={
          Object {
            "width": "14%",
          }
        }
      >
        0
        %
      </TableRowColumn>
    </TableRow>
    <TableRow
      displayBorder={true}
      hoverable={false}
      hovered={false}
      key="17"
      selectable={true}
      striped={false}
    >
      <TableRowColumn
        className="code"
        hoverable={false}
      >
        AAS.AX
      </TableRowColumn>
      <TableRowColumn
        className="company"
        hoverable={false}
      >
        THAILAND
      </TableRowColumn>
      <TableRowColumn
        className="price"
        hoverable={false}
      >
        88.87
      </TableRowColumn>
      <TableRowColumn
        className="value"
        hoverable={false}
      >
        <NumberFormat
          allowNegative={true}
          decimalSeparator="."
          displayType="text"
          isAllowed={[Function]}
          onChange={[Function]}
          onKeyDown={[Function]}
          onMouseUp={[Function]}
          prefix=""
          suffix=""
          thousandSeparator={true}
          type="text"
          value="38144159"
        />
      </TableRowColumn>
      <TableRowColumn
        className="increase"
        hoverable={false}
        style={
          Object {
            "width": "14%",
          }
        }
      >
        0
      </TableRowColumn>
      <TableRowColumn
        className="increase"
        hoverable={false}
        style={
          Object {
            "width": "14%",
          }
        }
      >
        0
        %
      </TableRowColumn>
    </TableRow>
    <TableRow
      displayBorder={true}
      hoverable={false}
      hovered={false}
      key="18"
      selectable={true}
      striped={false}
    >
      <TableRowColumn
        className="code"
        hoverable={false}
      >
        AAT.AX
      </TableRowColumn>
      <TableRowColumn
        className="company"
        hoverable={false}
      >
        IRAN
      </TableRowColumn>
      <TableRowColumn
        className="price"
        hoverable={false}
      >
        12.58
      </TableRowColumn>
      <TableRowColumn
        className="value"
        hoverable={false}
      >
        <NumberFormat
          allowNegative={true}
          decimalSeparator="."
          displayType="text"
          isAllowed={[Function]}
          onChange={[Function]}
          onKeyDown={[Function]}
          onMouseUp={[Function]}
          prefix=""
          suffix=""
          thousandSeparator={true}
          type="text"
          value="6827392"
        />
      </TableRowColumn>
      <TableRowColumn
        className="increase"
        hoverable={false}
        style={
          Object {
            "width": "14%",
          }
        }
      >
        0
      </TableRowColumn>
      <TableRowColumn
        className="increase"
        hoverable={false}
        style={
          Object {
            "width": "14%",
          }
        }
      >
        0
        %
      </TableRowColumn>
    </TableRow>
    <TableRow
      displayBorder={true}
      hoverable={false}
      hovered={false}
      key="19"
      selectable={true}
      striped={false}
    >
      <TableRowColumn
        className="code"
        hoverable={false}
      >
        AAU.AX
      </TableRowColumn>
      <TableRowColumn
        className="company"
        hoverable={false}
      >
        JAPAN
      </TableRowColumn>
      <TableRowColumn
        className="price"
        hoverable={false}
      >
        64.05
      </TableRowColumn>
      <TableRowColumn
        className="value"
        hoverable={false}
      >
        <NumberFormat
          allowNegative={true}
          decimalSeparator="."
          displayType="text"
          isAllowed={[Function]}
          onChange={[Function]}
          onKeyDown={[Function]}
          onMouseUp={[Function]}
          prefix=""
          suffix=""
          thousandSeparator={true}
          type="text"
          value="31927516"
        />
      </TableRowColumn>
      <TableRowColumn
        className="increase"
        hoverable={false}
        style={
          Object {
            "width": "14%",
          }
        }
      >
        0
      </TableRowColumn>
      <TableRowColumn
        className="increase"
        hoverable={false}
        style={
          Object {
            "width": "14%",
          }
        }
      >
        0
        %
      </TableRowColumn>
    </TableRow>
  </TableBody>
</Table>
`;
